### DISCUSSION TITLE
service singleton mode

### DISCUSSION THREAD

**Discussion by wu-clan** [fastapi-practices/fastapi_best_architecture#269] [Ideas]
Let's take a look at some of these great ideas.

**Comment by nipeone** [comment]
In fact, the CRUD layer is  the singleton pattern, I suggest that the Service layer and the CRUD adopt the same structure, so that the overall structure is  clearer

**Reply by wu-clan** [reply]
I'll think about it again. I'm considering whether to introduce ABC and add a PR later.

**Reply by nipeone** [reply]
introduce ABC is a good idea

**Reply by nipeone** [reply]
about celery, it is highly recommended to use folders to store multiple task files.

**Reply by wu-clan** [reply]
PR now available for viewing: #270

**Reply by nipeone** [reply]
why delete the pr?it can not make the proj more complex,otherwises, it's more clearer

**Reply by wu-clan** [reply]
Clearer, I agree, but more complex and probably unacceptable for pure python people.

Maybe a simple singleton pattern would suffice instead of ABC, but ABC feels good to me.

**Reply by wu-clan** [reply]
I reopened the PR and modified the structure.

**Reply by wu-clan** [reply]
Also to add, it seems to me that the CRUD method in the service cannot be generalized.

**Comment by wu-clan** [comment]
> about celery, it is highly recommended to use folders to store multiple task files.

Hi, @nipeone 

Since I'm not an expert in celery, it's a bit difficult for me in some of the design aspects.

**Reply by wu-clan** [reply]
Check this out, too. #237
